images/macos/provision/core/openjdk.sh

```diff
@@ -1,53 +1,94 @@
 #!/bin/bash -e -o pipefail
 source ~/utils/utils.sh
 
-JAVA_TOOLCACHE_PATH=$AGENT_TOOLSDIRECTORY/Java_Adopt_jdk
-
 createEnvironmentVariable() {
     local JAVA_VERSION=$1
-    local JAVA_PATH=$2
+    local VENDOR_NAME=$2
+    local DEFAULT=$3
 
-    local JAVA_HOME_PATH=$JAVA_PATH/Contents/Home
-    if [[ $JAVA_VERSION == $JAVA_DEFAULT ]]; then
-        echo "export JAVA_HOME=${JAVA_HOME_PATH}" >> "${HOME}/.bashrc"
+    if [[ ${DEFAULT} == "True" ]]; then
+        JAVA_PATH_PATTERN=$(eval echo ${AGENT_TOOLSDIRECTORY}/Java_${VENDOR_NAME}_jdk/${JAVA_VERSION}*/x64/Contents/Home/)
```

Comment:
duplicate path at line 10 and 15

Reply:
Why do we need eval?

Reply:
It's not a java path pattern

Reply:
@al-cheb eval here is needed to expand the version fully, our hosts' shell settings (setopt/shopt I supose) do not enable globs expansion by default, so whenever, say `$VERSION` holds `8` `$VERSION*`will always be represented as `8*` and never as a full version, might be @miketimofeev can comment on this matter more

Reply:
If I run a command `echo` command on a macOS hosted agent I will get the expanded value:
```
JAVA_VERSION=8
VENDOR_NAME="Adopt"
echo ${AGENT_TOOLSDIRECTORY}/Java_${VENDOR_NAME}_jdk/${JAVA_VERSION}*/x64/Contents/Home/
/Users/runner/hostedtoolcache/Java_Adopt_jdk/8.0.302-8/x64/Contents/Home
```

Reply:
@al-cheb oh, indeed, looks-like I've overengineered here